internal/cli/cmd/connectivity.go

```diff
@@ -77,16 +80,34 @@ func newCmdConnectivityTest() *cobra.Command {
 				return err
 			}
 
-			ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
-			defer cancel()
+			ctx, _ := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
 
 			go func() {
 				<-ctx.Done()
 				cc.Log("Interrupt received, cancelling tests...")
 			}()
 
-			if err := connectivity.Run(ctx, cc); err != nil {
-				fatalf("Connectivity test failed: %s", err)
+			done := make(chan struct{})
+			var finished bool
+
+			// Execute connectivity.Run() in its own goroutine, it might call Fatal()
+			// and end the goroutine without returning.
+			go func() {
+				defer func() { done <- struct{}{} }()
```

Comment:
nit: `close(done)` allows multiple goroutines to block on `done`. Not needed here of course.